### ISSUE TITLE
Lava and water blocks in c0.30 CPE

### ISSUE THREAD

**Issue by 4194304** [ViaVersion/ViaFabricPlus#152]
Water and lava buckets can only be accessed if you turn off the restricted inventory, and they do not turn into stone like other items.
Water and lava should not be present in the normal c0.30 inventory but should be present in the c0.30 CPE inventory since it is an extension, and also on many classic servers you can build with lava and water without it spreading everywhere.
Lava and water buckets on normal c0.30 turn into wool when placed, but on CPE they turn into lava and water as expected.
This would fit really nicely since the functionality is fully implemented, but is only accessible if you turn off a setting that affects gameplay.
Also if possible, create an option to disable the chat length limit in older versions, it is pretty irritating

**Comment by FlorianMichael** [comment]
What do you mean by chat length limitation? ViaFabricPlus should have the same limitation that the normal client also has

**Comment by 4194304** [comment]
what I mean by chat length limitation is how it limits how many characters you can type in the chat
![2023-06-24_13 42 07](https://github.com/ViaVersion/ViaFabricPlus/assets/130800536/57e01596-d62b-4bff-a077-37c6b7f870a9)

**Comment by 4194304** [comment]
Also as a bit of a side thought, have you ever thought about making brown wool turn into teal wool and black wool turn into violet wool? Teal wool displays as lime wool and violet wool displays as magenta wool, so there is no way to place teal wool or violet wool without ClassiCube or c0.30.

**Comment by FlorianMichael** [comment]
The chat length in the Classic should actually always be: 64 - username length - 2, unless the server is CPE and has LONGER_MESSAGES installed, then the Maximum length is Short.MAX_VALUE * 2 (65534)

**Comment by FlorianMichael** [comment]
> Water and lava buckets can only be accessed if you turn off the restricted inventory, and they do not turn into stone like other items. Water and lava should not be present in the normal c0.30 inventory but should be present in the c0.30 CPE inventory since it is an extension, and also on many classic servers you can build with lava and water without it spreading everywhere. Lava and water buckets on normal c0.30 turn into wool when placed, but on CPE they turn into lava and water as expected. This would fit really nicely since the functionality is fully implemented, but is only accessible if you turn off a setting that affects gameplay. Also if possible, create an option to disable the chat length limit in older versions, it is pretty irritating

I don't really understand what exactly is wrong with ViaFabricPlus, if it shouldn't be in a C0.30 inventory, then you shouldn't be able to place it at all, in ViaFabricPlus Water and lava are also only displayed as soon as the server has the extension.

**Comment by 4194304** [comment]
On CPE it should as it works properly and most servers fully support lava/water placement, also it is used widely for building on clients like Classicube, that's why

**Comment by 4194304** [comment]
ClassicCube mode gets rid of the chat length limit

**Comment by FlorianMichael** [comment]
> On CPE it should as it works properly and most servers fully support lava/water placement, also it is used widely for building on clients like Classicube, that's why

But I still don't understand concretely what should be changed now, should I just always allow Water/Lava and only translate to wool if the extension is missing?

**Comment by 4194304** [comment]
if it turns into wool hide it, if not and the server supports it, show it

**Comment by FlorianMichael** [comment]
But why should I add lava/water to the inventory if it's going to be remapped to wool anyway, then you should just take the wool right away?

**Comment by 4194304** [comment]
Also is it possible to get this to work https://github.com/ViaVersion/ViaFabricPlus/issues/152#issuecomment-1605725213

**Comment by 4194304** [comment]
might result in discoloration but it could be worth it if you made it change the wool textures depending on the version you chose

**Comment by FlorianMichael** [comment]
Why should I do that, it's completely random? I really do not understand what you want from me, I'm sorry.

**Comment by FlorianMichael** [comment]
> ClassicCube mode gets rid of the chat length limit

What do you mean by ClassiCube Mode? When you use the ClassiCube client or what?

**Comment by 4194304** [comment]
![Screenshot 2023-06-24 203438](https://github.com/ViaVersion/ViaFabricPlus/assets/130800536/5e0f4be5-41f0-4661-b49e-a1ff55dc7cb8)

**Comment by 4194304** [comment]
> Why should I do that, it's completely random? I really do not understand what you want from me, I'm sorry.

you can't place teal or violet wool

**Comment by FlorianMichael** [comment]
> ![Screenshot 2023-06-24 203438](https://user-images.githubusercontent.com/130800536/248526737-5e0f4be5-41f0-4661-b49e-a1ff55dc7cb8.png)

I will not remove the chat limit for the ClassiCube server list, because the chat limit is also in the normal ClassiCube client, otherwise it would be totally unlegit.

**Comment by FlorianMichael** [comment]
> > Why should I do that, it's completely random? I really do not understand what you want from me, I'm sorry.
> 
> you can't place teal or violet wool

Does the ClassiCube client not show the two blocks in the inventory or what?

**Comment by 4194304** [comment]
ClassiCube allows you to bypass the 64 character chat limit, teal wool displays as lime wool and violet wool appears as magenta wool

**Comment by FlorianMichael** [comment]
> ClassiCube allows you to bypass the 64 character chat limit, teal wool displays as lime wool and violet wool appears as magenta wool

Yes ClassiCube bypasses the chat limit, but only if the Longer Messages extension is installed on the server, and for that ViaFabricPlus has handling, that should remove the chat limit on servers that have LongerMessages installed, I just tested that again.

**Comment by 4194304** [comment]
Maybe OmniArchive's server is weird

**Comment by 4194304** [comment]
![classicube2](https://github.com/ViaVersion/ViaFabricPlus/assets/130800536/8d2ca455-73b4-48f5-95b4-7f7c2601e6bf)
![classicube](https://github.com/ViaVersion/ViaFabricPlus/assets/130800536/b776091b-c41b-43e7-8ed1-4d4dbd9b2b8e)
ClassiCube, correct wool colors, ViaFabricPlus, incorrect colors

**Comment by 4194304** [comment]
colors are repeated twice

**Comment by FlorianMichael** [comment]
It will take some time until this issue will be fixed here, because I don't have so much time in the near future, but there will be a new system for all old blocks and CPE blocks in the future with ViaFabricPlus 3.

**Comment by 4194304** [comment]
I thought that this issue was dead, but thank you for confirming that this could exist in ViaFabric 3
